File: src/main/java/com/lumiyaviewer/lumiya/render/shaders/RawShaderProgram.java
package com.lumiyaviewer.lumiya.render.shaders;

import android.opengl.GLES20;

public class RawShaderProgram extends ShaderProgram {
    public int textureSampler;
    public int uMVPMatrix;
    public int vPosition;
    public int vTexCoord;
    public int vTextureTransformMatrix;

    public RawShaderProgram(boolean z) {
        super(z ? Shader.ExtTextureVertexShader : Shader.RawVertexShader, z ? Shader.ExtTextureFragmentShader : Shader.RawFragmentShader);
    }

    protected void bindVariables() {
        this.vPosition = GLES20.glGetAttribLocation(this.handle, "vPosition");
        this.vTexCoord = GLES20.glGetAttribLocation(this.handle, "vTexCoord");
        this.uMVPMatrix = GLES20.glGetUniformLocation(this.handle, "uMVPMatrix");
        this.textureSampler = GLES20.glGetUniformLocation(this.handle, "vTexture");
        this.vTextureTransformMatrix = GLES20.glGetUniformLocation(this.handle, "vTextureTransformMatrix");
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/shaders/RawShaderProgram.kt
package com.lumiyaviewer.lumiya.render.shaders

import android.opengl.GLES20

/**
 * Basic shader program for raw texture rendering.
 * Supports both standard and external textures (for camera/video).
 */
class RawShaderProgram(useExternalTexture: Boolean) : ShaderProgram(
    if (useExternalTexture) Shader.ExtTextureVertexShader else Shader.RawVertexShader,
    if (useExternalTexture) Shader.ExtTextureFragmentShader else Shader.RawFragmentShader
) {
    var vPosition: Int = 0
    var vTexCoord: Int = 0
    var uMVPMatrix: Int = 0
    var textureSampler: Int = 0
    var vTextureTransformMatrix: Int = 0

    override fun bindVariables() {
        vPosition = GLES20.glGetAttribLocation(handle, "vPosition")
        vTexCoord = GLES20.glGetAttribLocation(handle, "vTexCoord")
        uMVPMatrix = GLES20.glGetUniformLocation(handle, "uMVPMatrix")
        textureSampler = GLES20.glGetUniformLocation(handle, "vTexture")
        vTextureTransformMatrix = GLES20.glGetUniformLocation(handle, "vTextureTransformMatrix")
    }
}
